src/snowflake/connector/ocsp_snowflake.py

```diff
@@ -695,14 +695,21 @@ def is_cache_fresh(current_time, ts):
         return current_time - OCSPCache.CACHE_EXPIRATION <= ts
 
     @staticmethod
-    def find_cache(ocsp, cert_id, subject, **kwargs):
+    def find_cache(
+        ocsp: SnowflakeOCSP, cert_id: CertId, subject: Certificate | None, **kwargs: Any
+    ):
         subject_name = ocsp.subject_name(subject) if subject else None
         current_time = int(time.time())
         cache_key: tuple[bytes, bytes, bytes] = kwargs.get(
             "cache_key", ocsp.decode_cert_id_key(cert_id)
         )
-        if cache_key in OCSP_RESPONSE_VALIDATION_CACHE:
-            ocsp_response_validation_result = OCSP_RESPONSE_VALIDATION_CACHE[cache_key]
+        lock_cache = kwargs.get("lock_cache", True)
```

Comment:
If you want to use `kwargs` I have a hard requirement that you need to type-hint these lines explicitly
```suggestion
        lock_cache: bool = kwargs.get("lock_cache", True)
```